internal/services/web/app_service_slot_custom_hostname_binding_resource.go

```diff
@@ -0,0 +1,193 @@
+	return &pluginsdk.Resource{
+		Create: resourceAppServiceSlotCustomHostnameBindingCreate,
+		Read:   resourceAppServiceSlotCustomHostnameBindingRead,
+		Delete: resourceAppServiceSlotCustomHostnameBindingDelete,
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.AppServiceSlotCustomHostnameBindingID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"app_service_slot_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.AppServiceSlotID,
+			},
+
+			"hostname": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"ssl_state": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(web.SslStateIPBasedEnabled),
+					string(web.SslStateSniEnabled),
+				}, false),
+				RequiredWith: []string{"thumbprint"},
+			},
+
+			"thumbprint": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				RequiredWith: []string{"ssl_state"},
+			},
+
+			"virtual_ip": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAppServiceSlotCustomHostnameBindingCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for App Service Slot Hostname Binding creation.")
+
+	slotId, err := parse.AppServiceSlotID(d.Get("app_service_slot_id").(string))
+	if err != nil {
+		return err
+	}
+
+	hostname := d.Get("hostname").(string)
+	sslState := d.Get("ssl_state").(string)
+	thumbprint := d.Get("thumbprint").(string)
+
+	id := parse.NewAppServiceSlotCustomHostnameBindingID(slotId.SubscriptionId, slotId.ResourceGroup, slotId.SiteName, slotId.SlotName, hostname)
+
+	locks.ByName(hostname, appServiceSlotCustomHostnameBindingResourceName)
+	defer locks.UnlockByName(hostname, appServiceSlotCustomHostnameBindingResourceName)
+
+	existing, err := client.GetHostNameBindingSlot(ctx, id.ResourceGroup, id.SiteName, id.SlotName, id.HostNameBindingName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+		}
+	}
+
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_app_service_slot_custom_hostname_binding", *existing.ID)
+	}
```

Comment:
can we update this to be:

```suggestion
	if !utils.ResponseWasNotFound(existing.Response) {
		return tf.ImportAsExistsError("azurerm_app_service_slot_custom_hostname_binding", id.ID())
	}
```

Reply:
fixed